### ISSUE TITLE
Invalid MEX-file '/home/hp/caffe/matlab/+caffe/private/caffe_.mexa64': /usr/lib/libgdal.so.1: undefined symbol: _ZN11xercesc_3_111InputSource11setEncodingEPKt

### ISSUE THREAD

**Issue by binialias** [BVLC/caffe#5984]
I am getting the below error when i issue the make mattest command

I am using the following version of softwares
ubuntu : 16.04
Matlab 2016a
gcc and g++ 4.9
OpenCV 3.2

Invalid MEX-file '/home/hp/caffe/matlab/+caffe/private/caffe_.mexa64':
/usr/lib/libgdal.so.1: undefined symbol:
_ZN11xercesc_3_111InputSource11setEncodingEPKt

Error in caffe.set_mode_cpu (line 5)
caffe_('set_mode_cpu');

Error in caffe.run_tests (line 6)
caffe.set_mode_cpu();
 
How to solve this issue?

**Comment by variantf** [comment]
https://github.com/BVLC/caffe/issues/3934#issuecomment-345719642

**Comment by tony-ch** [comment]
@binialias  Have you found a solution?

**Comment by 13293824182** [comment]
@tony-ch Have you found a solution?

**Comment by tony-ch** [comment]
@13293824182 I got the error when I used ubuntu 17.10 and matlab 2016a, now I use 16.04 and matlab 2015b, the error is gone.

**Comment by 13293824182** [comment]
@tony-ch   The problem didn't appear after the change? But I use 16.04 and matlab 2015b and i don't solve this problem?

**Comment by tony-ch** [comment]
@13293824182 Yes, the problem didn't appear after the change. Does the package "matlab-support" matter? I installed that package and there were no other changes.

**Comment by kjohew** [comment]
I have solved this problem by replace the OpenCV3.2 by OpenCV2.4. The reason is that opencv3.X call the libgdal.so and libgdal.so call system's libxercesc.so, which is conflict with the matlab's libxercesc.so, but opencv2.x doesn't need libgdal.so and will not lead this error. 

My software version is:
ubuntu : 17.10
Matlab 2016b
gcc and g++ 6.4
OpenCV2.4.13
